app/Http/Controllers/Backend/CategoryController.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace App\Http\Controllers\Backend;
+
+use Illuminate\Http\Request;
+
+use App\Http\Requests\Backend\CategoryRequest;
+use App\Http\Controllers\Controller;
+use App\Models\Backend\Category;
+use File;
+
+class CategoryController extends Controller
+{
+    /**
+     * Display a listing of the products.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $cates = Category::paginate(10);
+        $parent = Category::select('id', 'cate_name', 'parent_id')->get()->toArray();
+        return view('backend.categories.index', compact('cates', 'parent'));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        $cates = Category::select('id', 'cate_name', 'parent_id')->get()->toArray();
+        $list = new Category;
+        return view('backend.categories.create', compact('cates', 'list'));
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param \Illuminate\Http\Request $request request
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function store(CategoryRequest $request)
+    {
+        $addcate = new Category;
+        $addcate->cate_name = $request->cate_name;
+        $addcate->cate_description = $request->cate_description;
+        $addcate->cate_status = $request->cate_status;
+        $addcate->parent_id = $request->parent_id;
+        if ($request->hasFile('cate_image')) {
+            $fileName = $request->file('cate_image')->getClientOriginalName();
+            $addcate->cate_image = $fileName;
+            $request->file('cate_image')->move(public_path('upload/'), $fileName);
+        }
+        $addcate->save();
+        $request->session()->flash('message', 'Category was add successfully!');
+        return redirect()->route('admin.categories.index');
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param int $id id
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
+    {
+        $data = Category::findOrFail($id);
+        $cates = Category::select('id', 'cate_name', 'parent_id')->get()->toArray();
+        $list = new Category;
```

Comment:
[rank C] what's this line meaning? I don't understand why do you create new object there and by pass it to view.